### ISSUE TITLE
Setting black/white lists of entrypoints in a provider 

### ISSUE THREAD

**Issue by athoune** [traefik/traefik#5848]
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?

I don't understand why a dynamic provider can mess with my private entrypoint (something like 8080 on a private interface).

A setting like `allowed_entrypoints`, used with a logical and should be enough.

A black list should be another answer for the same question.

**Comment by traefiker** [comment]
Hi! I'm Træfiker :robot: the bot in charge of communication regulation.

Thanks for your interest in Traefik!

Issue templates help us help you by providing all necessary information.

Please edit your issue and use the available templates:

- [Feature request](https://raw.githubusercontent.com/containous/traefik/master/.github/ISSUE_TEMPLATE/Feature_request.md)
- [Bug report](https://raw.githubusercontent.com/containous/traefik/master/.github/ISSUE_TEMPLATE/Bug_report.md)

And remember: each time someone ignores the template, a cute little bunny dies.
<details>
<summary>#SaveTheCuteBunny :heart: :rabbit: :heart:</summary>

![southpark - estrella kill rabbit](https://user-images.githubusercontent.com/5674651/27143914-3cea7896-5130-11e7-81f8-02d8ac2da7a7.gif)

![southpark - estrella kill rabbit2](https://user-images.githubusercontent.com/5674651/27143924-49104c72-5130-11e7-82ae-ccedbd5dfa1c.gif)

</details>

**Comment by athoune** [comment]
bunny saved.

**Comment by dduportal** [comment]
Hi @athoune , thanks for the saved bunny ;) 

To be honest, I'm not sure to understand the issue, could you elaborate a bit please?

- What do you mean by `a dynamic provider can mess with my private entrypoint`?
- What would be the behavior of `A setting like allowed_entrypoints` ?
- When asking for a black list, it would be a list of what exactly?

Thanks a lot!

**Comment by athoune** [comment]
I'm following the documentation, and I set 3 entry points : *web* (0.0.0.0:80), *websecure* (0.0.0.0:443) and *traefik* (localhost:8080).

When I start some docker containers with minimum labels, without specifying entrpoints, the default behavior is "use all the entrypoints". I don't want dynamic config for my private *traefik* entrypoint, I don't want docker service here. I want explicit admin stuff on *traefik* entrypoint, written in a config file.

The actual rule : "when a dynamic service doesn't specify entrpoints, use all anetrypoints" is useful.
But I want to be able to isolate some entrypoints from dynamic configuration.
Entrypoints should be isolated with a withelist, or a blacklist (xor, both should be confusing).

A setting named `allowed_entrypoints` inside the provider block, should specify the array of allowed entrypoints. The dynamic service can only ask for a subset of this setting.

**Comment by bradjones1** [comment]
> A setting named `allowed_entrypoints` inside the provider block, should specify the array of allowed entrypoints. The dynamic service can only ask for a subset of this setting.

Yes. Or, put another way, restrict the entrypoints on which a particular provider's services are configured.

**Comment by fenduru** [comment]
In my case this would be useful as a means of preventing my docker services from ever being exposed via HTTP instead of HTTPS. While I have a web entrypoint that redirects to websecure, Docker service routers still get generated on the web entrypoint. Assuming the priorities are correct this is fine, but it introduces a potential issue if priorities are changed (since one of the docker routers could end up with a higher priority than the redirect router)

**Comment by nmengin** [comment]
Hello there,

This proposal has been open for a while with minimal traction. We won't be adding it to our roadmap at this time and prefer to close it for now.

BTW, concerning the issue with traefik entrypoint exposure, we've fixed it.

If circumstances change or interest grows, we can always reopen this later.
Thank you, @athoune , for the suggestion!
